clippy_lints/src/returns.rs

```diff
@@ -253,29 +275,21 @@ fn check_final_expr<'tcx>(
     }
 }
 
-fn emit_return_lint(
-    cx: &LateContext<'_>,
-    ret_span: Span,
-    semi_spans: Vec<Span>,
-    inner_span: Option<Span>,
-    replacement: RetReplacement,
-) {
+fn expr_contains_if<'tcx>(expr: &'tcx Expr<'tcx>) -> bool {
+    match expr.kind {
+        ExprKind::If(..) => true,
+        ExprKind::Binary(_, left, right) => expr_contains_if(left) || expr_contains_if(right),
+        _ => false,
+    }
```

Comment:
This will now also suggest brackets, if only a single if statement is present, like

```
fn b() -> u32 {
    return if true {1} else {2};
}
```

Could you adjust the recursion in a way, that it only returns true if they're nested in binary expressions? I think the simplest option would be to pass in a bool value `on_if` that is returned for `ExprKind::If(..)`

Could you than also add the example above as a test?

Reply:
Thank you for pointing that out~ your are right, my code missed that case.